plugins/janus_recordplay.c

```diff
@@ -2007,6 +2008,33 @@ static void *janus_recordplay_handler(void *data) {
 			}
 			/* Tell the core to tear down the PeerConnection, hangup_media will do the rest */
 			gateway->close_pc(session->handle);
+		} else if (!strcasecmp(request_text, "pause") || !strcasecmp(request_text, "resume")) {
+			JANUS_LOG(LOG_VERB, "Record&Play: Got pause/resume request\n");
+			if(session->recording) {
+				gboolean pause = !strcasecmp(request_text, "pause");
```

Comment:
I think this should probably check what the state of the recording is, before doing something. As it is, a "pause" on a paused recording will set the paused property again, and will send a new event too, which seems incorrect. I think this is particolarly important due to the way "resume" is implemented in record.c, where the last time is reset every time the method is called.

I guess the right way to do that would be a new property on the `session->recording` object: maybe a volatile int to check and tweak via `g_atomic_int_compare_and_exchange` as we do in a few other places.